app/controllers/api/account.php

```diff
@@ -99,8 +100,10 @@
                 '$write' => ['user:' . $userId],
                 'email' => $email,
                 'emailVerification' => false,
+                'phone' => $phone,
+                'phoneVerification' => false,
                 'status' => true,
-                'password' => Auth::passwordHash($password, Auth::DEFAULT_ALGO, Auth::DEFAULT_ALGO_OPTIONS),
+                'password' => (!empty($password)) ? Auth::passwordHash($password, Auth::DEFAULT_ALGO, Auth::DEFAULT_ALGO_OPTIONS) : '',
```

Comment:
In cases where we want to pass an empty password, we do it with a `null` rather than an empty string.

https://github.com/appwrite/appwrite/blob/bb4693ddfc154fe01f617a811fbc71af926b74d2/app/controllers/api/account.php#L1015

Reply:
This code has been removed.